tests/rptest/scale_tests/tiered_storage_reader_stress_test.py

```diff
@@ -0,0 +1,319 @@
+            ResourceSettings(memory_mb=4096, num_cpus=2))
+
+    def _produce_and_quiesce(self, topic_name: str, msg_size: int,
+                             data_size: int, expect_bandwidth: float,
+                             **kwargs):
+        expect_runtime = max(60.0, (data_size / expect_bandwidth) * 2)
+
+        t1 = time.time()
+        timeout = expect_runtime * self.runtime_grace_factor
+
+        self.logger.info(f"Producing {data_size} bytes, timeout = {timeout}")
+        KgoVerifierProducer.oneshot(self.test_context,
+                                    self.redpanda,
+                                    topic_name,
+                                    msg_size=msg_size,
+                                    msg_count=data_size // msg_size,
+                                    batch_max_bytes=msg_size * 8,
+                                    timeout_sec=timeout,
+                                    **kwargs)
+        produce_duration = time.time() - t1
+        self.logger.info(
+            f"Produced {data_size} bytes in {produce_duration} seconds, {(data_size/produce_duration)/1000000.0:.2f}MB/s"
+        )
+
+        quiesce_uploads(
+            self.redpanda, [topic_name],
+            self.manifest_upload_interval + self.segment_upload_interval + 30)
+
+    def _create_topic(self, topic_name: str, partition_count: int,
+                      local_retention: int):
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic(topic_name,
+                         partitions=partition_count,
+                         replicas=3,
+                         config={
+                             'retention.local.target.bytes': local_retention,
+                         })
+
+    def _get_stats(self):
+        """The stats we care about for reader stress, especially
+        reader count and kafka connection count"""
+
+        results = {}
+        for node in self.redpanda.nodes:
+            metrics = self.redpanda.metrics(
+                node, metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+            segment_reader_count = 0
+            partition_reader_count = 0
+            partition_reader_delay_count = 0
+            connection_count = 0
+            for family in metrics:
+                for sample in family.samples:
+                    if sample.name == "redpanda_cloud_storage_readers":
+                        segment_reader_count += int(sample.value)
+                    if sample.name == "redpanda_cloud_storage_partition_readers":
+                        partition_reader_count += int(sample.value)
+                    if sample.name == "redpanda_cloud_storage_partition_readers_delayed_total":
+                        partition_reader_delay_count += int(sample.value)
+                    elif sample.name == "redpanda_rpc_active_connections":
+                        if sample.labels["redpanda_server"] == "kafka":
+                            connection_count += int(sample.value)
```

Comment:
This will be very interesting, given the rare shutdown hangs in CI that appear to maybe be caused by lingering Kafka connections